mautic/mautic
# Add filter grouping to reporting

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Related user documentation PR URL | Pending
| Related developer documentation PR URL | Pending
| Issues addressed (#s or URLs) | Pending
| BC breaks? | None expected
| Deprecations? | N/A

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

This PR adds the long awaited feature to be able to add grouping to report filters. In other words, you can now build reports with `contact first name equals Don OR contact last name equals Gilbert`. Current report filters join all filters as an `AND`. This means that report filter functionality now mirrors segment filter functionality.

Note: If you're testing on PHP 7.1, please apply #4246 as well as this

[//]: # ( As applicable: )
#### Steps to test this PR:
1. Apply PR, clear cache, and rebuild assets via `app/console m:a:g`
2. Go to reports, and create a new Contact report
3. Add a filter on `Contact first name` equals the first name of one of your contacts
4. Add a second filter on contact first name, and make it equal the first name of another contact, but select `OR` as the glue.
5. Save the report, and see that both contacts are brought in
